docs/advanced-features/middleware.md

```diff
@@ -1,105 +1,247 @@
 ---
-description: Learn how to use Middleware in Next.js to run code before a request is completed.
+description: Learn how to use Middleware to run code before a request is completed.
 ---
 
-# Middleware (Beta)
+# Middleware
 
 <details open>
-  <summary><b>Version History</b></summary>
+<summary><b>Version History</b></summary>
 
-| Version   | Changes                                                                                                 |
-| --------- | ------------------------------------------------------------------------------------------------------- |
-| `canary`  | Preparing for stability, see [upgrade guide](https://nextjs.org/docs/messages/middleware-upgrade-guide) |
-| `v12.0.9` | Enforce absolute URLs in Edge Runtime ([PR](https://github.com/vercel/next.js/pull/33410))              |
-| `v12.0.0` | Middleware (Beta) added.                                                                                |
+| Version | Changes |
+
+| --------- | ------------------------------------------------------------------------------------------ |
+
+| `v12.2.0` | Middleware GA |
+
+| `v12.0.9` | Enforce absolute URLs in Edge Runtime ([PR](https://github.com/vercel/next.js/pull/33410)) |
+
+| `v12.0.0` | Middleware (Beta) added |
 
 </details>
 
-Middleware enables you to run code before a request is completed. Based on the user's incoming request, you can modify the response by rewriting, redirecting, adding headers, or even streaming HTML.
+Middleware allows you to run code before a request is completed, then based on the incoming request, you can modify the response by rewriting, redirecting, adding headers, or setting cookies.
+
+When a request is made, it will first hit the Middleware, _then_ the cache, meaning you can personalize static content and implement authentication, run A/B tests, deliver personalizes pages based on geolocation, and perform bot protection.
```

Comment:
```suggestion
When a request is made, it will first hit the Middleware, _then_ the cache, meaning you can personalize static content and implement authentication, run A/B tests, deliver personalized pages based on geolocation, and perform bot protection.
```